### PR TITLE
terminal & git inside pair storm

### PR DESCRIPTION
**Pull request by PeterLysyk**

### CODE REVIEW

File: src/bottompanel/bottompaneldock.cpp [hunk 1]
```diff
@@ -1,30 +1,38 @@
 #include "bottompaneldock.h"
-
 #include <QTabWidget>
-
 #include "mainwindow.h"
+#include "consolewindow/consolewindow.h"
+#include <QDebug>
```
Comment (opastushkov) [inline src/bottompanel/bottompaneldock.cpp:5]: Remove qDebug

File: src/bottompanel/consolewindow/consoleserviceprovider.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#include "consoleserviceprovider.h"
+#include "consoleliterals.h"
+#include "documentmanager.h"
+#include <QTextCodec>
+#include <QDebug>
```
Comment (opastushkov) [inline src/bottompanel/consolewindow/consoleserviceprovider.cpp:5]: Remove qDebug

File: src/bottompanel/consolewindow/consoleserviceprovider.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+#include <QtCore>
```
Comment (opastushkov) [inline src/bottompanel/consolewindow/consoleserviceprovider.cpp:6]: QtCore is too huge. Include only useful modules.

File: src/bottompanel/consolewindow/consoleserviceprovider.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+void ConsoleServiceProvider::setCompilerPath(const QString &compilerPath)
+{
+    mCompilerPath = compilerPath;
+}
+
+void ConsoleServiceProvider::setStdOutput()
+{
+    if (QSysInfo::productType() == windowsProductType)// if it's windows console
```
Comment (opastushkov) [inline src/bottompanel/consolewindow/consoleserviceprovider.cpp:28]: Use one comment style.

File: src/bottompanel/consolewindow/consoleserviceprovider.cpp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+void ConsoleServiceProvider::runConsoleCommand(QString command)
+{
+    QString executionCommand;
+    if (QSysInfo::productType() == windowsProductType)
+    {
+        executionCommand = cmdWindowsStartPath;
+    }
+    executionCommand +=command;
```
Comment (opastushkov) [inline src/bottompanel/consolewindow/consoleserviceprovider.cpp:47]: ```executionCommand += command;```

File: src/bottompanel/bottompaneldock.h [hunk 6]
```diff
@@ -14,6 +15,15 @@ class BottomPanelDock: public QDockWidget
     QTabWidget *mpTabWgt;
 public:
     explicit BottomPanelDock(QWidget *pParent = nullptr);
+signals:
+    void projectPathWasChanged(QString);
+public slots:
+    void reSendProjectPathChanged(QString);
```
Comment (opastushkov) [inline src/bottompanel/bottompaneldock.h:21]: ```void resendProjectPathChanged(QString);```

File: src/bottompanel/consolewindow/consoleliterals.h [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#ifndef CONSOLELITERALS_H
+#define CONSOLELITERALS_H
+
+#include <QString>
```
Comment (opastushkov) [inline src/bottompanel/consolewindow/consoleliterals.h:4]: You included QString and don't use it.

File: src/bottompanel/consolewindow/consolewindow.cpp [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+ConsoleWindow::ConsoleWindow(QWidget *parent) :
+    QWidget(parent),
+    ui(new Ui::ConsoleWindow)
+{
+    ui->setupUi(this);
+    consoleServiceProvider = new ConsoleServiceProvider;
+    consoleView = new ConsoleView;
+
+    QGridLayout *consoleViewGridLayout = new QGridLayout (this);
```
Comment (opastushkov) [inline src/bottompanel/consolewindow/consolewindow.cpp:14]: ```QGridLayout *consoleViewGridLayout = new QGridLayout(this);```

File: src/bottompanel/consolewindow/consolewindow.h [hunk 9]
```diff
@@ -0,0 +1,28 @@
+#ifndef CONSOLEWINDOW_H
+#define CONSOLEWINDOW_H
+
+#include <QWidget>
+#include "consoleserviceprovider.h"
+#include "consoleview.h"
+
+namespace Ui
+{
+class ConsoleWindow;
```
Comment (opastushkov) [inline src/bottompanel/consolewindow/consolewindow.h:10]: Add tab

File: src/mainwindow/mainwindow.h [hunk 10]
```diff
@@ -42,10 +42,11 @@ class MainWindow : public QMainWindow
     ProjectViewerDock *mpProjectViewerDock;
     ChatWindowDock *mpChatWindowDock;
     BottomPanelDock *mpBottomPanelDock;
-    QScopedPointer<DocumentManager> mpDocumentManager;    
+    //QScopedPointer<DocumentManager> mpDocumentManager;
```
Comment (opastushkov) [inline src/mainwindow/mainwindow.h:45]: Remove commented code.

**Review by opastushkov — CHANGES_REQUESTED** [review]

File: src/bottompanel/consolewindow/consoleserviceprovider.cpp (hunk 4 above)
Comment (PeterLysyk) [reply src/bottompanel/consolewindow/consoleserviceprovider.cpp:28]: What did you mean?

File: src/bottompanel/consolewindow/consolewindow.h (hunk 9 above)
Comment (PeterLysyk) [reply src/bottompanel/consolewindow/consolewindow.h:10]: QT generated this by itself

**Pull request closed without merging**